### PR TITLE
guile-fibers: improve build

### PR DESCRIPTION
**Pull request by leungbk**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/guile-modules/guile-fibers/default.nix [hunk 1]
```diff
@@ -2,29 +2,40 @@
...
     owner = "wingo";
     repo = "fibers";
     rev = "v${version}";
-    hash = "sha256-kU/ty/XRNfv3ubIwH40wZmo8MXApeduHcH2KEGqoh+Q=";
+    hash = "sha256-jEY6i+uTqDkXZKdpK+/GRLlK7aJxkRneVZQJIE4bhlI=";
   };
 
+  patches = [
+    ./guile-fibers-epoll-instance-is-dead.patch
+    ./guile-fibers-wait-for-io-readiness.patch
```
Comment (marsam) [inline pkgs/development/guile-modules/guile-fibers/default.nix:24]: Please use `fetchpatch` for these patches.
Besides that, LGTM.

**Comment by leungbk** [comment]
I decided not to init the `guile-goblins` package after all since my recipe wasn't very clean, and making it clean would require some kind of `guile3.withPackages` construct that I am too lazy to write.

The other commits should be OK though.

**Comment by SuperSandro2000** [comment]
From your commit message I get what you did but not why. We could surely merge the three commits into one with a short description what this improves. I think cross compilation is better supported after this.

**Comment by wegank** [comment]
Closed by #247510.

**Pull request closed without merging**